ext/storage_sources/s3_store/test/test_list_buckets.cpp

```diff
@@ -0,0 +1,61 @@
+#include <aws_bucket_conn.h>
+#include <aws/core/Aws.h>
+
+#include <iostream>
+
+/* Default config settings for the S3CrtClient. */
+namespace test_defaults 
+{
+    const Aws::String region = Aws::Region::AP_SOUTHEAST_2;
+    const double throughput_target_gbps = 5;
+    const uint64_t part_size = 8 * 1024 * 1024; /* 8 MB. */
+}
+
+/* Wrapper for unit test functions. */
+#define TEST(func, config, expected)                   \
+    do {                                               \
+        int __ret;                                     \
+        if ((__ret = (func(config))) != expected)      \
+            return (__ret);                            \
+    } while (0)
+
+/*
+ * test_list_buckets --
+ *     Unit test for listing S3 buckets under the associated AWS account.
+ */
+int 
```

Comment:
Is the test considered to be successful is the list is empty?

Reply:
This test eventually won't be required - listing buckets is not a functionality that AWSBucketConn will need to support. it's currently more of an example test to show that CTest is working as intended and that the tests can access the AWS SDK.

Reply:
It would be great to add what you just said to the header of the test then.